experimental/SetPartitions/src/SetPartition.jl

```diff
@@ -0,0 +1,293 @@
+"""
+    SetPartition
+
+A set-partition is a partition of a set of upper and lower points into disjoint subsets. 
+Such partitions are often depicted as string diagrams where points in the same subset 
+are connected by lines. See also Section 4.1.1 in [Gro20](@cite).
+
+`SetPartition` stores two vectors `upper_points` and `lower_points`
+where same numbers correspond to points in the same subset. Both vectors are 
+automatically converted into normal form as returned by [`normal_form`](@ref).
```

Comment:
This is about internals of the struct. Users shouldn't touch/see/modify them, so this information is useless to misleading in a docstring. If you want to give users a way to see/modify this stuff, you should write appropriate functions.